### PR TITLE
Initial code considerations and other sections

### PR DESCRIPTION
**Pull request by KodrAus**
Contributes to #1 and #3 

This is a large drop of our existing forge docs that's been re-organized into a _Code considerations_ section. It's got a stubbed out reviewer checklist too that we could try turn into a friendly document that reviewers can refer to when PRs come along.

I've also stubbed out a _Lifecycle of a feature_ section where we can talk about new unstable APIs, tracking issues, stabilizations, and deprecations.

It's all still very rough 🙂 But should give us space to start capturing more code guidelines and start documenting our processes.

r? @m-ou-se @Mark-Simulacrum

### CODE REVIEW

**Comment by KodrAus** [comment]
Also cc @jyn514 and @camelid who've expressed an early interest in the std dev guide.

**Comment by KodrAus** [comment]
The process sections are still _very_ sparse, but we should be able to fill them out pretty well.

File: src/tools-and-bots/crater.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# `@craterbot`
+
+**Status:** Stub
+
+[Crater](https://github.com/rust-lang/crater) is a tool that can test PRs against a public subset of the Rust ecosystem to estimate the scale of potential breakage.
+
+You can kick off a crater run with:
+
+```
+@bors try @craterbot check
```
Comment (m-ou-se) [inline src/tools-and-bots/crater.md:10]: Does this work? I was under the impression that unlike rust-timer, craterbot doesn't wait for a `bors try` and can only be started after the `bors try` build has finished.

**Review by m-ou-se — APPROVED** [review]
Nice! Looking forward to filling out all this stuff more. :)

File: src/tools-and-bots/crater.md (hunk 1 above)
Comment (jyn514) [reply src/tools-and-bots/crater.md:10]: I think you're right and it has to be separate. It might be nice to extend craterbot to do this, though.

Comment (KodrAus) [reply src/tools-and-bots/crater.md:10]: 😢

**Comment by KodrAus** [comment]
I'll go ahead and merge this in then so we can start working on it!

**Pull request merged**